circe/src/main/scala/org/http4s/circe/middleware/JsonDebugErrorHandler.scala

```diff
@@ -0,0 +1,140 @@
+package org.http4s.circe.middleware
+
+import cats._
+
+import cats.effect._
+import cats.data._
+import io.circe._
+import io.circe.syntax._
+import org.http4s._
+import org.http4s.headers.{Connection, `Content-Length`}
+import org.http4s.util.CaseInsensitiveString
+import org.http4s.implicits._
+import org.http4s.circe._
+
+object JsonDebugErrorHandler {
+
+  private[this] val messageFailureLogger =
+    org.log4s.getLogger("org.http4s.circe.jsondebugerrorhandler.message-failures")
```

Comment:
Nit: It's a bit weird to me that this doesn't match the package whence it comes.